### DISCUSSION TITLE
Problems when use 'to_scipy_sparse_matrix'

### DISCUSSION THREAD

**Discussion by Byun-jinyoung** [pyg-team/pytorch_geometric#4284] [Q&A] [answered]
Dear pyg uesrs,
I've tried to process my input data with torch_geometic `to_scipy_sparse_matrix`.

Below is an example of a my graph
`Data(x=[6593, 42], y=[1], pos=[6593, 3], bond_edge_index=[2, 16410], bond_edge_attr=[16410, 2], nbond_edge_index=[2, 210975], nbond_edge_attr=[210975, 2])`

And my code is that
```
edge_index, edge_attr = get_laplacian(
            graph.nbond_edge_index, graph.nbond_edge_attr, normalization='sym',
            num_nodes=graph.num_nodes)
L = to_scipy_sparse_matrix(edge_index, edge_attr).tocsc()
```
The edge_index and edge_attr after laplacian have
`edge_index torch.Size([2, 217568]) edge attr torch.Size([217568, 2])`

But I got the error code at `L = to_scipy_sparse_matrix(edge_index, edge_attr).tocsc()` as below
```
File "/home/byun/anaconda3/envs/pyg_torch/lib/python3.7/site-packages/torch_geometric/utils/convert.py", line 34, in to_scipy_sparse_matrix
    assert edge_attr.size(0) == row.size(0)
AssertionError
```
But, in pyg document, it is described that edge_attr (Tensor, optional) – Edge weights or multi-dimensional edge features. (default: None)

Could you tell me what I'm missing...?

Best regards, 
Thank you:)

**Comment by rusty1s** [comment, marked as answer]
`scipy` can only hold one-dimensional edge features inside its sparse matrices. As such, you would need to create two matrices (one for each edge feature):
```python
L1 = to_scipy_sparse_matrix(edge_index, edge_attr[:, 0]).tocsc()
L2 = to_scipy_sparse_matrix(edge_index, edge_attr[:, 1]).tocsc()
```

**Reply by Byun-jinyoung** [reply]
Oh Thank you!
